test/test-integration/src/test/java/org/wso2am/micro/gw/tests/util/HttpsPostMultipart.java

```diff
@@ -0,0 +1,134 @@
+     * @param headers
+     * @throws IOException
+     */
+    public HttpsPostMultipart(String requestURL, Map<String, String> headers) throws IOException {
+
+        HttpsClientRequest.setSSlSystemProperties();
+
+        boundary = UUID.randomUUID().toString();
+        URL url = new URL(requestURL);
+        conn = (HttpsURLConnection) url.openConnection();
+
+        conn.setUseCaches(false);
+        conn.setDoOutput(true);    // indicates POST method
+        conn.setDoInput(true);
+        conn.setReadTimeout(30000);
+        conn.setConnectTimeout(15000);
+        conn.setDoInput(true);
+
+        conn.setAllowUserInteraction(false);
+        conn.setRequestProperty("Content-Type", "multipart/form-data; boundary=" + boundary);
+        if (headers != null && headers.size() > 0) {
+            Iterator<String> it = headers.keySet().iterator();
+            while (it.hasNext()) {
+                String key = it.next();
+                String value = headers.get(key);
+                conn.setRequestProperty(key, value);
+            }
+        }
+        outputStream = conn.getOutputStream();
+        writer = new PrintWriter(new OutputStreamWriter(outputStream,TestConstant.CHARSET_NAME), true);
+    }
+
+
+    /**
+     * Adds a form field to the request
+     *
+     * @param name  field name
+     * @param value field value
+     */
+    public void addFormField(String name, String value) {
+        writer.append("--" + boundary).append(LINE);
+        writer.append("Content-Disposition: form-data; name=\"" + name + "\"").append(LINE);
+        writer.append("Content-Type: text/plain; charset=" + TestConstant.CHARSET_NAME).append(LINE);
+        writer.append(LINE);
+        writer.append(value).append(LINE);
+        writer.flush();
+    }
+
+    /**
+     * Adds a upload file section to the request
+     *
+     * @param fieldName
+     * @param uploadFile
+     * @throws IOException
+     */
+    public void addFilePart(String fieldName, File uploadFile)
+            throws IOException {
+        String fileName = uploadFile.getName();
+        writer.append("--" + boundary).append(LINE);
+        writer.append("Content-Disposition: form-data; name=\"" + fieldName + "\"; filename=\"" + fileName + "\"").append(LINE);
+        writer.append("Content-Type: " + URLConnection.guessContentTypeFromName(fileName)).append(LINE);
+        writer.append("Content-Transfer-Encoding: binary").append(LINE);
+        writer.append(LINE);
+        writer.flush();
+
+        FileInputStream inputStream = new FileInputStream(uploadFile);
+        byte[] buffer = new byte[4096];
+        int bytesRead = -1;
+        while ((bytesRead = inputStream.read(buffer)) != -1) {
+            outputStream.write(buffer, 0, bytesRead);
+        }
+        outputStream.flush();
+        inputStream.close();
+        writer.append(LINE);
+        writer.flush();
+    }
+
+    /**
+     * Completes the request and receives response from the server.
+     *
```

Comment:
Shall we add the java docs for return type here?

Reply:
Fixed in 3b6cc6755810d748cd3de4fb25696a5692c08e78